data_pipeline/gmail_production_data/get_gmail_data.py

```diff
@@ -0,0 +1,114 @@
+import pandas as pd
+import numpy as np
+
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+
+
+# this connection type will be changed !!
+def connect_to_email(userId):
+    '''
+    To connect gmail account via service account
+    param:
+        - userId : email adress
+    return: None
+    '''
+    SCOPES = ['https://www.googleapis.com/auth/gmail.readonly']
+    SERVICE_ACCOUNT_FILE = 'DELETE/elife-data-pipeline-test.json'
+    EMAIL_ADDRESS = userId
+
+    credentials = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+    delegated_credentials = credentials.with_subject(EMAIL_ADDRESS)
+
+    service = build("gmail", "v1", credentials=delegated_credentials)
+    return service
+
+
+def get_label_list(service, userId):
```

Comment:
I would suggest to make use of Python's typing hints (and we could check them using `mypy`).
Personally, I try to add as little as possible code comments. Some comments may be a considered a "code smell". I can't find a good link but this is one of the results: https://softwareengineering.stackexchange.com/questions/1/comments-are-a-code-smell

Just as an example, not to critique your comment (because the same would be true for my comments sooner or later): the comment is out of sync as it is not writing to a CSV file. The parameter list is incomplete and if we added the type hints would just repeat the same information.

i.e. still add comments that you think are useful but not in the code. But in some cases we could change the code to include that information.

One information that is not be as easy to read from the code, but might be relevant is what fields does this actually return, as it's not just a list of strings.
I guess in this case the fields of https://developers.google.com/gmail/api/reference/rest/v1/users.labels#Label
I suppose the labels().list method already has a reference to it, but I personally would find it useful (although not necessary) here with maybe the main fields "id" and "name.